lib/spack/llnl/util/filesystem.py

```diff
@@ -1133,21 +1135,31 @@ def lexists_islink_isdir(path):
     return True, is_link, is_dir
 
 
+class BaseDirectoryVisitor(object):
+    """Base class and interface for visit_directory_tree."""
+
+    def visit_file(self, root, rel_path, depth):
+        pass
```

Comment:
All these methods could benefit docstring to make the semantic of the function and input arguments explicit.

Reply:
done